[DOCU-2838] Konnect System Accounts (GA) (#4999)
Kong/docs.konghq.com: app/konnect/org-management/system-accounts.md

---
title: Manage System Accounts
content_type: how-to
no_version: true
---

This guide explains what a system account is, how they vary from user accounts, and how to manage a system account using the [{{site.konnect_short_name}} API](https://developer.konghq.com/spec/5175b87f-bfae-40f6-898d-82d224387f9b/fc735302-d8ac-4e66-a9ef-225569a75d3c). 

System accounts are a service account in {{site.konnect_short_name}}. Because they are not associated with an email address and a user, system accounts can be used for automation and integrations. 

System accounts offer the following benefits over regular user accounts:

* System accounts are not associated with an email address. This allows you to use the account as part of an automation or integration that isn't associated with any person’s identity.
* When you use a user account as part of an automation or integration and that user leaves the company, this breaks the automations and integrations. If you use a system account instead, the automations and integrations wouldn't break.

> * Because system accounts can't be assigned the same permissions as a user account, they can't access the {{site.konnect_short_name}} UI.

Reviewer: ```suggestion
* System accounts do not have sign in credentials and thus cannot access the {{site.konnect_short_name}} UI. These accounts are intended for use with APIs and CLIs.
```

Suggestion: SA's can be assigned the same roles as user accounts so I think the reasoning is incorrect. We should point out that they are for use with API and CLI.

## Manage a system account via the API

### Create a system account

Create a system account by sending a `POST` request containing the `name` of your system account in the response body:

```sh
curl --request POST \
  --url https://global.api.konghq.com/v2/system-accounts
  --data '{
  "name": "Example System Account"}'
```

You will receive a `201` response code, and a response body containing information about your system account:

```
{
  "id": "497f6eca-6276-4993-bfeb-53cbbbba6f08",
  "name": "Example System Account",
  "created_at": "2019-08-24T14:15:22Z",
  "updated_at": "2019-08-24T14:15:22Z"
}
```

### Generate a system account access token

The system account access token can be used for authenticating API and CLI requests.

Create a system account by sending a `POST` request containing the `accountId` of the system account:

```sh
curl --request POST \
  --url https://global.api.konghq.com/v2/system-accounts/:497f6eca-6276-4993-bfeb-53cbbbba6f08/access-tokens
```
You will receive a `201` response code, and a response body containing the access token for the system account:

```
{
  "id": "497f6eca-6276-4993-bfeb-53cbbbba6f08",
  "name": "Sample Access Token",

>   "created_at": "2019-08-24T14:15:22Z",
>   "updated_at": "2019-08-24T14:15:22Z",
>   "expires_at": "2019-08-24T14:15:22Z",
>   "last_used_at": "2019-08-24T14:15:22Z",

Reviewer: ```suggestion
  "created_at": "2023-01-12:15:54Z",
  "expires_at": "2023-11-15T00:00:00Z",
  "updated_at": "2023-01-13T21:04:22Z",
  "last_used_at": "2023-01-18T06:45:40Z",
```

More realistic example.

>   "token": "npat_12345678901234567890123456789012345678901234567890"

Reviewer: ```suggestion
  "token": "spat_12345678901234567890123456789012345678901234567890"
```

The correct prefix for system account tokens are `spat_`

}
```

{:.important}
> **Important**: The access token is only displayed once, so make sure you save it securely. 

### Assign a role to a system account

You can assign a role to a system account so that the permissions associated with that role can be assigned to that account and their subsequent credentials.

Assign a role to a system account by sending a `POST` request containing the `accountId` and the `role_name` of the system account:

```sh
curl --request POST \
  --url https://global.api.konghq.com/v2/system-accounts/:497f6eca-6276-4993-bfeb-53cbbbba6f08/assigned-roles

>   --data '{
>   "role_name": "Viewer"}'

Reviewer: ```suggestion
  --data '{
  "role_name": "Viewer",
  "entity_id": "817d0422-45c9-4d88-8d64-45aef05c1ae7",
  "entity_type_name": "Runtime Groups",
  "entity_region": "eu"
}'
```

The system account can use a {{site.konnect_short_name}} personal access token (PAT) the same way a [regular {{site.konnect_short_name}} user](/konnect/org-management/users/) can. In addition, the system account can be assigned roles directly or inherit the roles of a [team](/konnect/org-management/teams-and-roles/). As such, a PAT created by a system account inherits the roles assigned to the system account.

## Manage a system account via the API

### Create a system account

Create a system account by sending a `POST` request containing the `name` of your system account in the response body:

```sh
curl --request POST \
  --url https://global.api.konghq.com/v2/system-accounts
  --data '{
  "name": "Example System Account"}'
```

You will receive a `201` response code, and a response body containing information about your system account:

```json
{
  "id": "497f6eca-6276-4993-bfeb-53cbbbba6f08",
  "name": "Example System Account",
  "created_at": "2019-08-24T14:15:22Z",
  "updated_at": "2019-08-24T14:15:22Z"
}
```

### Generate a system account access token

The system account access token can be used for authenticating API and CLI requests.

> Create a system account by sending a `POST` request containing the `accountId` of the system account:

Reviewer 2: ```suggestion
Create a system account token by sending a `POST` request containing the `accountId` of the system account:
```

---
title: Manage System Accounts
content_type: how-to
no_version: true
---

> This guide explains what a system account is, how they vary from user accounts, and how to manage a system account using the [{{site.konnect_short_name}} API](https://developer.konghq.com/spec/5175b87f-bfae-40f6-898d-82d224387f9b/fc735302-d8ac-4e66-a9ef-225569a75d3c). 

Reviewer 2: ```suggestion
This guide explains what a system account is, how it varies from a user account, and how to manage a system account using the [{{site.konnect_short_name}} API](https://developer.konghq.com/spec/5175b87f-bfae-40f6-898d-82d224387f9b/fc735302-d8ac-4e66-a9ef-225569a75d3c). 
```
Fixing singular/plural agreement.

The system account can use a {{site.konnect_short_name}} personal access token (PAT) the same way a [regular {{site.konnect_short_name}} user](/konnect/org-management/users/) can. In addition, the system account can be assigned roles directly or inherit the roles of a [team](/konnect/org-management/teams-and-roles/). As such, a PAT created by a system account inherits the roles assigned to the system account.

## Manage a system account via the user interface

You can create a system account using the {{site.konnect_short_name}} user interface by navigating to **Organization > System Accounts**. You can also assign system accounts to teams and manage system account tokens from this page.

## Manage a system account via the API

### Create a system account

Create a system account by sending a `POST` request containing the `name` of your system account in the response body:

```sh
curl --request POST \
  --url https://global.api.konghq.com/v2/system-accounts
  --data '{
  "name": "Example System Account"}'
```

You will receive a `201` response code, and a response body containing information about your system account:

> ```

Reviewer 2: ```suggestion
```json
```

{
  "id": "497f6eca-6276-4993-bfeb-53cbbbba6f08",
  "name": "Example System Account",
  "created_at": "2019-08-24T14:15:22Z",
  "updated_at": "2019-08-24T14:15:22Z"
}
```

### Generate a system account access token

The system account access token can be used for authenticating API and CLI requests.

Create a system account by sending a `POST` request containing the `accountId` of the system account:

```sh
curl --request POST \
  --url https://global.api.konghq.com/v2/system-accounts/:497f6eca-6276-4993-bfeb-53cbbbba6f08/access-tokens
```
You will receive a `201` response code, and a response body containing the access token for the system account:

> ```

Reviewer 2: ```suggestion
```json
```

{
  "id": "497f6eca-6276-4993-bfeb-53cbbbba6f08",
  "name": "Sample Access Token",
  "created_at": "2023-01-12:15:54Z",
  "expires_at": "2023-11-15T00:00:00Z",
  "updated_at": "2023-01-13T21:04:22Z",
  "last_used_at": "2023-01-18T06:45:40Z",
  "token": "spat_12345678901234567890123456789012345678901234567890"
}
```

> 

Reviewer 2: ```suggestion

Copy and save the access token beginning with `spat_`.

```

{:.important}
> **Important**: The access token is only displayed once, so make sure you save it securely. 

### Assign a role to a system account

You can assign a role to a system account so that the permissions associated with that role can be assigned to that account and their subsequent credentials.

Assign a role to a system account by sending a `POST` request containing the `accountId` and the `role_name` of the system account:

```sh
curl --request POST \
  --url https://global.api.konghq.com/v2/system-accounts/:497f6eca-6276-4993-bfeb-53cbbbba6f08/assigned-roles
  --data '{
  "role_name": "Viewer",
  "entity_id": "817d0422-45c9-4d88-8d64-45aef05c1ae7",
  "entity_type_name": "Runtime Groups",
  "entity_region": "eu"
}'
```

You will receive a `201` response code and a response body containing the role that is now assigned to the system account:

> ```

Reviewer 2: ```suggestion
```json
```

{
  "id": "497f6eca-6276-4993-bfeb-53cbbbba6f08",
  "role_name": "Viewer",
  "entity_id": "817d0422-45c9-4d88-8d64-45aef05c1ae7",
  "entity_type_name": "Runtime Groups",
  "entity_region": "eu"
}
```

### Assign a system account to a team

You can assign a team to a system account so that the permissions associated with that team can be assigned to that account and their subsequent credentials.

Assign a team to a system account by sending a `POST` request containing the `teamId` of the team:

> ```sh

Author: ```suggestion
```json
```

System accounts offer the following benefits over regular user accounts:

* System accounts are not associated with an email address. This allows you to use the account as part of an automation or integration that isn't associated with any person’s identity.
* When you use a user account as part of an automation or integration and that user leaves the company, automation and integrations break. If you use a system account instead, the automation and integrations wouldn't break.
* System accounts don't have sign-in credentials and therefore can't access the {{site.konnect_short_name}} UI. These accounts are intended to be used with APIs and CLIs.

The system account can use a {{site.konnect_short_name}} personal access token (PAT) the same way a [regular {{site.konnect_short_name}} user](/konnect/org-management/users/) can. In addition, the system account can be assigned roles directly or inherit the roles of a [team](/konnect/org-management/teams-and-roles/). As such, a PAT created by a system account inherits the roles assigned to the system account.

> ## Manage a system account via the user interface
> 
> You can create a system account using the {{site.konnect_short_name}} user interface by navigating to **Organization > System Accounts**. You can also assign system accounts to teams and manage system account tokens from this page.
> 

Reviewer: ```suggestion
```
Remove the UI mention until UI becomes available.

Storing the original content here for when UI becomes available.
```
## Manage a system account via the user interface

You can create a system account using the {{site.konnect_short_name}} user interface by navigating to **Organization > System Accounts**. You can also assign system accounts to teams and manage system account tokens from this page.
```
